src/soter/openssl/soter_ec_key_3.c

```diff
@@ -0,0 +1,452 @@
+/*
+* Copyright (c) 2023 Cossack Labs Limited
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#ifdef THEMIS_EXPERIMENTAL_OPENSSL_3_SUPPORT
+
+#include "soter/soter_ec_key.h"
+
+#include <string.h>
+
+#include <openssl/bn.h>
+#include <openssl/core_names.h>
+#include <openssl/evp.h>
+#include <openssl/obj_mac.h>
+#include <openssl/opensslv.h>
+#include <openssl/param_build.h>
+
+#include "soter/openssl/soter_engine.h"
+#include "soter/soter_portable_endian.h"
+
+static bool is_curve_supported(const char* curve)
```

Comment:
reviewed one more time these functions and looks like we should write 1 function that maps curve name to curve integer ID used in openssl 1.x, and re-use existing functions. it does less copy-paste and more efficient checks of input due to simple comparison of integers instead of strcmp. We can move such functions to `soter_ec_key_common.c` and use it for both files `soter_ec_key.c`/`soter_ec_key_3.c`